src/main/java/io/nats/client/api/ConsumerConfiguration.java

```diff
@@ -128,9 +134,20 @@ public String toJson() {
         JsonUtils.addField(sb, RATE_LIMIT_BPS, rateLimit);
         JsonUtils.addFieldAsNanos(sb, IDLE_HEARTBEAT, idleHeartbeat);
         JsonUtils.addField(sb, FLOW_CONTROL, flowControl);
+        if (maxPullWaiting != 0 && maxPullWaiting != DEFAULT_MAX_PULL_WAITING) {
```

Comment:
Don't present the zero or default value to the server. Might reconsider and only block zero.